tests/vmi_hook_sidecar_test.go

```diff
@@ -218,6 +223,87 @@ var _ = Describe("[sig-compute]HookSidecars", decorators.SigCompute, func() {
 			})
 		})
 
+		Context("with sidecar-shim", func() {
+			It("should receive Terminal signal on VMI deletion", func() {
+				vmi = tests.RunVMIAndExpectLaunch(vmi, 360)
+
+				err = virtClient.VirtualMachineInstance(testsuite.GetTestNamespace(vmi)).Delete(context.Background(), vmi.Name, &metav1.DeleteOptions{})
+				Expect(err).ToNot(HaveOccurred())
+
+				Eventually(func() bool {
+					vmiPod, exists, err := getVMIPod(vmi)
+					if err != nil {
+						Expect(err).NotTo(HaveOccurred(), "must be able to retrieve the VMI virt-launcher pod")
```

Comment:
This is weird, remove guard around `Expect(err).NotTo(HaveOccurred())`?

Reply:
Also, if you Expect inside Eventually you most likely want to do it like this:

```
Eventually(func(g Gomega) {
    g.Expect(err).ToNot(HaveOccured())
...
```

Otherwise you fail the whole test instead of running the Eventually func multiple times, what is probably what you intended to do.